### PR TITLE
Update tool versions and dependencies

### PR DESCRIPTION
**Pull request by bestbeforetoday**
- Remove several unnecessary dependencies from Java package.
- Update Makefile to latest toolchain and to work on Arm64 Mac.
- Add security vulnerability scan.
- Update dependencies to address CVE-2023-2976

### CODE REVIEW

File: .github/workflows/vulnerability-scan.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - uses: actions/setup-java@v3
+        with:
+          java-version: 17
+          distribution: temurin
+          cache: maven
+      - name: Scan
+        run: make scan-java
+      - name: "Archive dependency-check report"
```
Comment (jt-nti) [inline .github/workflows/vulnerability-scan.yml:53]: Should Go and node also upload a scan report?

**Review by jt-nti — APPROVED** [review]

File: .github/workflows/vulnerability-scan.yml (hunk 1 above)
Comment (bestbeforetoday) [reply .github/workflows/vulnerability-scan.yml:53]: They don’t generate any report file, only outputting to the console, which can be viewed in the build log. Only the Java scan actually outputs a separate report with more information than output to the console.

Comment (jt-nti) [reply .github/workflows/vulnerability-scan.yml:53]: Aha, that makes sense then! :)

**Comment by bestbeforetoday** [comment]
Use OSV-Scanner (on a generated SBOM) instead of the Maven dependency-check plugin, since OSV-Scanner seems to require less maintenance and produces fewer false positives.

**Comment by bestbeforetoday** [comment]
@denyeart Please review this and merge once you are happy. It addresses security vulnerabilities so it will need to be back-ported to 0.2.x (for Fabric v2.5 LTS) and a release published to allow client APIs to pick it up.

File: bindings/go-apiv2/go.mod [hunk 2]
```diff
@@ -3,14 +3,14 @@ module github.com/hyperledger/fabric-protos-go-apiv2
 go 1.17
 
 require (
-	google.golang.org/grpc v1.46.2
-	google.golang.org/protobuf v1.28.0
+	google.golang.org/grpc v1.57.0
```
Comment (denyeart) [inline bindings/go-apiv2/go.mod:6]: Fabric core repository is having an issue with the grpc update, I meant to look into that before merging this one, let me go back and look into that further...

Comment (bestbeforetoday) [reply bindings/go-apiv2/go.mod:6]: This should not impact Fabric core. This is an update to fabric-protos-apiv2, which is not used by Fabric core. fabric-gateway does use this package, but fabric-gateway is also using the same gRPC version and tests are passing fine:

https://github.com/hyperledger/fabric-gateway/blob/320d90ca584b59cf5628b01420403ffbc5cd7b61/go.mod#L14

Comment (denyeart) [reply bindings/go-apiv2/go.mod:6]: Ok cool, will merge.
I did find the issue with grpc update in core fabric so I will resolve that separately.

**Review by denyeart — APPROVED** [review]

**Pull request merged**